src/test/java/org/opensearch/neuralsearch/query/HybridBulkScorerTests.java

```diff
@@ -106,4 +106,107 @@ public void testScoreWithInvalidRange() throws IOException {
+     * Verifies that the minimum position is always returned.
+     *
+     * MaxDocId 1000 and window size is 4096
+     */
+    public void testGetNextDocIdCandidate_whenMaxDocIdInSegmentIsLowerThanWindowSize() throws IOException {
+        // Create 3 scorers
+        Scorer mockScorer3 = mock(Scorer.class);
+        DocIdSetIterator mockIterator3 = mock(DocIdSetIterator.class);
+        when(mockIterator3.cost()).thenReturn(1L);
+        when(mockScorer3.iterator()).thenReturn(mockIterator3);
+
+        List<Scorer> scorers = Arrays.asList(mockScorer1, mockScorer2, mockScorer3);
+        HybridBulkScorer bulkScorer = new HybridBulkScorer(scorers, true, MAX_DOC);
+
+        LeafCollector mockLeafCollector = mock(LeafCollector.class);
+
+        // Setup: scorer1 at 500, scorer2 at 100, scorer3 at 300
+        when(mockIterator1.docID()).thenReturn(-1);
+        when(mockIterator1.advance(0)).thenReturn(500);
+        when(mockIterator1.nextDoc()).thenReturn(DocIdSetIterator.NO_MORE_DOCS);
+
+        when(mockIterator2.docID()).thenReturn(-1);
+        when(mockIterator2.advance(0)).thenReturn(100);
+        when(mockIterator2.nextDoc()).thenReturn(DocIdSetIterator.NO_MORE_DOCS);
+
+        when(mockIterator3.docID()).thenReturn(-1);
+        when(mockIterator3.advance(0)).thenReturn(300);
+        when(mockIterator3.nextDoc()).thenReturn(DocIdSetIterator.NO_MORE_DOCS);
+
+        when(mockScorer1.score()).thenReturn(1.0f);
+        when(mockScorer2.score()).thenReturn(0.8f);
+        when(mockScorer3.score()).thenReturn(0.9f);
+
+        // Execute scoring
+        int result = bulkScorer.score(mockLeafCollector, null, 0, 4096);
+
+        assertEquals("All Iterators reach the end", Integer.MAX_VALUE, result);
+    }
+
+    /**
+     * Test getNextDocIdCandidate when all scorers are exhausted.
+     * Should return NO_MORE_DOCS.
+     */
+    public void testGetNextDocIdCandidateAllScorersExhausted() throws IOException {
+        List<Scorer> scorers = Arrays.asList(mockScorer1, mockScorer2);
+        HybridBulkScorer bulkScorer = new HybridBulkScorer(scorers, true, MAX_DOC);
+
+        LeafCollector mockLeafCollector = mock(LeafCollector.class);
+
+        // Setup: both scorers exhausted
+        when(mockIterator1.docID()).thenReturn(-1);
+        when(mockIterator1.advance(0)).thenReturn(DocIdSetIterator.NO_MORE_DOCS);
+        when(mockIterator1.nextDoc()).thenReturn(DocIdSetIterator.NO_MORE_DOCS);
+
+        when(mockIterator2.docID()).thenReturn(-1);
+        when(mockIterator2.advance(0)).thenReturn(DocIdSetIterator.NO_MORE_DOCS);
+        when(mockIterator2.nextDoc()).thenReturn(DocIdSetIterator.NO_MORE_DOCS);
+
+        // Execute scoring
+        int result = bulkScorer.score(mockLeafCollector, null, 0, MAX_DOC);
+
+        // Should return NO_MORE_DOCS when all scorers are exhausted
+        assertEquals("Should return NO_MORE_DOCS when all scorers exhausted", DocIdSetIterator.NO_MORE_DOCS, result);
+    }
```

Comment:
we need more test cases, but that can be added in later PRs:
1. multi-window iteration test
test that multiple windows are processed correctly with advancing doc IDs:
2. window boundary edge cases
3. sparse scorer (don't confuse with sparse query)  test

Reply:
ack.